### DISCUSSION TITLE
Contributors are welcome to have more regular release

### DISCUSSION THREAD

**Discussion by mcuee** [libusb/libusb#1325] [Announcements]
I think many people will agree libusb is an important open source project. We are short of maintainers and contributors. If you are interested to be a maintainer, please let us know. 

If you are not ready to be a maintainer, you can still contribute. For example, you can review the pull requests, you can participate in the issue discussions in github and you can join the libusb-devel mailing list discussions.

**Comment by mcuee** [comment]
Within the three main OS platforms, Linux is pretty mature, macOS is also in pretty good shape but Windows is lagging behind due to the complexities.

But you do not need to know Windows codes well enough to join the maintainer team. 

For example, handling the release can already help. Right now only Nathan and Chris are actively handling the release.

Release since 2015.
libusb 1.0.24: by Chris on 11 Dec 2020
libusb 1.0.23 by Nathan on on 29 Aug 2019
libusb 1.0.22 by Chris on 25 Mar 2018
libusb 1.0.21 by Nathan on 26 Oct 2016
libusb 1.0.20 by Chris on 14 Sep 2015

**Comment by mcuee** [comment]
Pull requests are very much welcome.

For example, even within Windows issues, there are easier ones and more complicated ones (eg: hotplug, isoc related).
https://github.com/libusb/libusb/issues?q=is%3Aopen+is%3Aissue+label%3Awindows

**Comment by mcuee** [comment]
Reviewers/testers for the pull requests are also very welcome.

For example, for Android it seems to me #874 is moving towards the right direction. So it will be good to review and test $874 if you care for Android.

**Comment by mcuee** [comment]
BTW, I spent a bit of time recently to label all the tickets.

As of today (10 July 2021), we have closed 151 items related to Windows over the years, 60 items for macOS, 56 for Linux and 27 for Android. We still have 20 items open for Windows, 2 open for macOS, 1 open for Linux and 5 for Android.

**Comment by mcuee** [comment]
With Tormod Volden on board, thanks to his great efforts we have release 1.0.25 and 1.0.26.

libusb 1.0.25: released by Tormod on 01 Feb 2022
Update: 1.0.26: released by Tormod on 10 April 2022

We will welcome more people to come on board.

**Reply by Mr-Bossman** [reply]
@mcuee how are the release made? I would like to add Windows ARM.

**Reply by mcuee** [reply]
We need the agreement from the main developers for a realeas.

Windows ARM binaries are not generated in CI as of now. You may want to start with a PR for that.

**Reply by mcuee** [reply]
For 1.0.28 release --> please check this issue out.
* https://github.com/libusb/libusb/issues/1453

**Reply by Mr-Bossman** [reply]
Oh oh I meant generate the windows release under the releases tab on GH

**Reply by mcuee** [reply]
Windows release binary is built using the CI build. So you need to create a PR to generate Windows ARM binary first.

**Reply by tormodvolden** [reply]
There is an issue for it now: #1600
If it gets built in CI we may offer it as part of the release packages. But we would still encourage developers to compile libusb themselves.

**Comment by mcuee** [comment]
For testing of libusb
https://github.com/libusb/libusb/wiki/FAQ#Testing_libusb

**Comment by mcuee** [comment]
Please take note of the use of ChatGPT or Github Copilot or similar AI tools.

https://github.com/libusb/libusb/pull/1327#issuecomment-1770297698
* > Because of the unclear copyright situation I am not comfortable with merging stuff coming straight out of ChatGPT or code copilot etc (as long as I know it, which I of course cannot unless people are frank about it) into a GPL project.

**Comment by mcuee** [comment]
libusb-1.0.27 has been released by @tormodvolden in Jan 2024.
https://github.com/libusb/libusb/releases/tag/v1.0.27

Hopefully we can get more reviewers and testers to get libusb-1.0.28 release in year 2024.
* https://github.com/libusb/libusb/issues/1453

**Reply by mcuee** [reply]
Unfortunately that did not happen. I think we really need more developers and reviewers/testers.

**Comment by mcuee** [comment]
Let's welcome @sonatique to the admin team.

@sonatique is one of the few persons who know about libusb Windows. I believe he will be a very good addition to the libusb team.

**Comment by mcuee** [comment]
libusb-1.0.29 has been released.

**Comment by mcuee** [comment]
Now we have @seanm in the admin team.

**Comment by minicarpet** [comment]
I saw that you are really searching for some people, I can help as maintainer, on my side we are using Libusb for Windows and Linux with custom device (0xFF class).

**Comment by mcuee** [comment]
We will need more contributors for review and testing of existing PRs, looking through issues and potentially sort out the issues with solutions including PRs.

Platforms we may have to rely more on the community to help on the following platforms, in terms of both testing and enhancement.
1) Android (I will need to learn how to carry out test on Android. This is the one platform I have not really touched).
2) OpenBSD and NetBSD
3) Solaris/illumos based OS
4) Haiku OS
5) WebAssembly

More on the testng side.
1) Windows on ARM64
2) Maybe some exotic Linux distros for not-so-common architectures

We do not really drop OS support unless necessary (eg: we have to drop support of Windows XP). On the other hand, we do not have the capability/capacity to carry out the test on legacy platforms. So if you really need to use legacy versions and find issues, you have to raise PRs for review. We may accept the PR if it does not affect the supported platforms.

1) Legacy macOS versions (macports has versions back to Snow Leopard 10.6 for i386 and x86_64)
https://ports.macports.org/port/libusb/details/

2) Legacy Windows versions like Windows Vista (??), Windows 7 and even Windows 10, if you really need to use them.

3) Legacy Linux OS

**Comment by mcuee** [comment]
libusb-1.0.30 has been released.
https://github.com/libusb/libusb/releases/tag/v1.0.30 (18-May-2026)

BTW, I have also released libusb-compat-0.1.9 on the same date.
https://github.com/libusb/libusb-compat-0.1/releases/tag/v0.1.9

**Comment by Mr-Bossman** [comment]
Is it possible for non-project-member contributors to request reviews from specific people?

**Reply by Youw** [reply]
You can always tag someone specific in a comment.

**Reply by Mr-Bossman** [reply]
kk, but it clutters the comments.
